src/mod/applications/mod_voicemail/mod_voicemail.c

```diff
@@ -829,11 +886,13 @@ static vm_profile_t *get_profile(const char *profile_name)
 
 static void profile_rwunlock(vm_profile_t *profile)
 {
-	switch_thread_rwlock_unlock(profile->rwlock);
-	if (switch_test_flag(profile, PFLAG_DESTROY)) {
-		if (switch_thread_rwlock_trywrlock(profile->rwlock) == SWITCH_STATUS_SUCCESS) {
-			switch_thread_rwlock_unlock(profile->rwlock);
-			free_profile(profile);
+	if (profile) {
```

Comment:
lets simplify changes to:

if (!profile) return;

Reply:
No problem ! Done !